lib/glific_web/resolvers/sheets.ex

```diff
@@ -56,6 +56,22 @@ defmodule GlificWeb.Resolvers.Sheets do
     end
   end
 
+  @doc false
+  @spec sync_sheet(Absinthe.Resolution.t(), %{id: integer, input: map()}, %{context: map()}) ::
+          {:ok, any} | {:error, any}
+  def sync_sheet(_, %{id: id}, %{context: %{current_user: user}}) do
+    with {:ok, sheet} <-
+           Repo.fetch_by(Sheet, %{id: id, organization_id: user.organization_id}),
+         {:ok, sheet} <-
+           Sheets.update_sheet(sheet, %{
+             url: sheet.url,
+             label: sheet.label,
+             organization_id: user.organization_id
+           }) do
+      {:ok, %{sheet: sheet}}
+    end
```

Comment:
Did not get it. what are you trying to do here?

Reply:
I'm using the sheet update function which adds `last_synced_at` in the update_sheet function and update the `sheet data` table and deletes sheet data that is not present in the sheet after recent sync

Reply:
No no @AkhileshNegi that's bad practice and the code looks too coupled. 

Also, let's say if you just update the sheer label will it resync the complete sheet again? Also, you can add more columns like user roles, shortcode and updating them will perform the same operation.

So, please take out that function from the update_sheet and create a new one called sync_with_google or something. 

So whenever you will get a request to update sheet data you can update the data and call that sync function after that if needed. 

Also last_synced_at will always be updated form sync function and not the update one.

Reply:
@pankaj-ag made requested changes